Python/ceval.c

```diff
@@ -1097,9 +1101,12 @@ _PyEval_EvalFrameDefault(PyFrameObject *f, int throwflag)
                 */
                 goto fast_next_opcode;
             }
-            if (_Py_atomic_load_relaxed(&pendingcalls_to_do)) {
+            if ((next_instr > handle_pending_after) &&
+                    _Py_atomic_load_relaxed(&pendingcalls_to_do)) {
                 if (Py_MakePendingCalls() < 0)
                     goto error;
+                /* Allow for subsequent jumps backwards in the bytecode */
+                handle_pending_after = first_instr;
```

Comment:
Is it possible to get a jump backwards without the `&pendingcalls_to_do` becoming true in the mean time, so that this never gets reset?

Reply:
Yeah, it is, so the check should be restructured a bit to handle that case.

Reply:
Maybe the simplest fix would be for the frame to track a start and stop (with the start being implicitly set to the offset of the `DEFER_PENDING_UNTIL` opcode), and make the check `if (_Py_atomic_load_relaxed(&pendingcalls_to_do) && !(defer_pending_start <= next_instr && next_instr < defer_pending_end)`. Using a standard half-open range here would also make it easy to say "there is currently no defer pending region": just set `defer_pending_start = defer_pending_end`.

Reply:
Yeah, that would work, and means I can put the check back inside the `pending_calls_to_do` guard again.